velox/exec/HashBuild.cpp

```diff
@@ -166,15 +169,26 @@ void HashBuild::setupSpiller(SpillPartition* spillPartition) {
   if (!spillEnabled()) {
     return;
   }
-
   const auto& spillConfig = spillConfig_.value();
   HashBitRange hashBits = spillConfig.hashBitRange;
-  if (spillPartition != nullptr) {
+
+  if (spillPartition == nullptr) {
+    spillGroup_->addOperator(
+        *this,
+        [&](const std::vector<Operator*>& operators) { runSpill(operators); });
+  } else {
+    spillInputReader_ = spillPartition->createReader();
+
     const auto startBit = spillPartition->id().partitionBitOffset() +
         spillConfig.hashBitRange.numBits();
+    // Disable spilling if exceeding the max spill level.
```

Comment:
Curious what is the error message on exceeding spill level. Does it explain what happened clearly?

Reply:
It will run out of memory if the restored partition can't fit in-memory. Comment added.

Reply:
Would it be confusing to the user having spilling enabled yet still running out of memory. Would it make sense to improve the error message to explain why spilling couldn't help?